hexa/plugins/connector_s3/models.py

```diff
@@ -455,25 +482,39 @@ def update_metadata(self, object_data):
 
         self.key = object_data["true_key"]
         self.parent_key = self.compute_parent_key(object_data["true_key"])
-        self.size = object_data["size"]
-        self.storage_class = object_data["StorageClass"]
+        if "size" in object_data:
+            self.size = object_data["size"]
+        elif "ContentLength" in object_data:
+            self.size = object_data["ContentLength"]
+        else:
+            raise KeyError("size")
+        self.storage_class = object_data.get("StorageClass", "STANDARD")
         self.type = object_data["type"]
         self.last_modified = object_data.get("LastModified")
         self.etag = object_data.get("ETag")
+        if self.etag and self.etag.startswith('"'):
+            self.etag.strip('"')
         self.orphan = False
 
     @classmethod
     def create_from_object_data(cls, bucket, object_data):
+        if "size" in object_data:
+            size = object_data["size"]
+        elif "ContentLength" in object_data:
+            size = object_data["ContentLength"]
+        else:
+            raise KeyError("size")
+
         # TODO: move to manager
         return cls.objects.create(
             bucket=bucket,
             key=object_data["true_key"],
             parent_key=cls.compute_parent_key(object_data["true_key"]),
-            size=object_data["size"],
-            storage_class=object_data["StorageClass"],
-            type=object_data["type"],
+            storage_class=object_data.get("StorageClass", "STANDARD"),
             last_modified=object_data.get("LastModified"),
-            etag=object_data.get("ETag"),
+            etag=object_data["ETag"].strip('"') if "ETag" in object_data else None,
```

Comment:
Curious to hear about this - maybe worth a comment?

Reply:
same as L184 it turns out. Except it isn't a bug in S3FS, it's the real API return. I didn't find any mention of this in docs.
Example: {
...
'ETag': '"195c1cb9a00c10b54d5a1000363fb0cb"',
...
}